src/composition/watchers/subgraphs.rs

```diff
@@ -382,16 +373,33 @@ impl SubgraphHandles {
         // and propagate the update through by forcing a recomposition. This may be unnecessary,
         // but we'll figure that out on the receiving end rather than passing around more
         // context.
+        let routing_url = lazily_resolved_subgraph.routing_url().clone().or(async move {
```

Comment:
__NIT__: `.or()` will cause the async closure to always be executed and awaited, even if we don't need it

Reply:
Yer that makes sense, thanks for flagging, have pushed a fix :)